mlflow/data/delta_dataset_source.py

```diff
@@ -13,6 +26,11 @@
 # it is neither a UC nor local metastore catalog
 DATABRICKS_SAMPLES_CATALOG_NAME = "samples"
 
+_ACTIVE_CATALOG_QUERY = "SELECT current_catalog() AS catalog"
```

Comment:
Can we reuse code from elsewhere in MLflow for this? e.g. https://github.com/mlflow/mlflow/blob/ae27dedccd7d3b60ce79859c7755196d0e282083/mlflow/store/_unity_catalog/registry/utils.py#L29 instead of duplicating it

Reply:
I'll need to pull all of that out into a common util then as I don't think we should be importing from store/_unity_catalog/... here.